sympy/core/mul.py

```diff
@@ -1255,19 +1255,20 @@ def _eval_is_integer(self):
             return False
 
         n, d = fraction(self)
-        if not is_rational:
-            _self = n/d
-            if _self != self:
-                return _self.is_integer
         if is_rational:
-            if d is S.One:
+            if d == S.One:
                 return True
             elif d == S(2):
                 return n.is_even
-        # if d is even -- 0 or not -- the
-        # result is not an integer
         if n.is_odd and d.is_even:
```

Comment:
If these are true then rational must be true so this should be under that clause.

Reply:
`is_rational` didn't work for all cases.
I'll post a couple of cases soon